### PR TITLE
Add example for NFS provisioner

### LINKED ISSUE #55: create an nfs provisioner example
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

there is already a nfs server container here: https://github.com/kubernetes-csi/csi-driver-nfs/blob/master/test/nfs-testdriver.go#L113

could you follow this doc: https://github.com/kubernetes-csi/csi-driver-smb/tree/master/deploy/example/smb-provisioner, to set up a similar nfs server

I think we need to set up an NFS server provisioner example  first, that would be easy for testing

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by mayankshah1607**
Signed-off-by: Mayank Shah <mayankshah1614@gmail.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #55

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: examples/kubernetes/nfs-provisioner/pv-nfs-csi.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+metadata:
+  name: pv-nfs
+spec:
+  capacity:
+    storage: 1Gi
+  accessModes:
+    - ReadWriteMany
+  persistentVolumeReclaimPolicy: Retain
+  mountOptions:
+    - dir_mode=0777
```
Comment (andyzhangx) [inline examples/kubernetes/nfs-provisioner/pv-nfs-csi.yaml:13]: i don't think these mountOptions would work. may use `vers=3`

File: examples/kubernetes/nfs-provisioner/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Set up a NFS Server on a Kubernetes cluster
+
+### 1. Create a NFS Server Deployment
+
+```bash
+kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/nfs-driver-smb/master/examples/kubernetes/nfs-provisioner/nfs-server.yaml
+```
+
+### 2. Create a PersistentVolume
```
Comment (andyzhangx) [inline examples/kubernetes/nfs-provisioner/README.md:9]: PV is not part of this nfs-provisioner config, may move this example to other place

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

File: examples/kubernetes/nfs-provisioner/README.md (hunk 2 above)
Comment (mayankshah1607) [reply examples/kubernetes/nfs-provisioner/README.md]: Sure, I have removed it for now. Added example for getting public IP using LoadBalancer instead

Comment (andyzhangx) [reply examples/kubernetes/nfs-provisioner/README.md]: pls use `ClusterIP` by default, `LoadBalancer` requires cloud provider, let's not depend on cloud provider in this project

Comment (mayankshah1607) [reply examples/kubernetes/nfs-provisioner/README.md]: Yes the `nfs-server.yaml` uses ClusterIP. `nfs-server-lb.yaml` uses LB. Shall I remove that example or keep it optional?

Comment (andyzhangx) [reply examples/kubernetes/nfs-provisioner/README.md]: keep it as optional is fine

Comment (mayankshah1607) [reply examples/kubernetes/nfs-provisioner/README.md]: Sure, the updated readme mentions nfs-server.yaml as default. I could make changes there if required, thanks

File: examples/kubernetes/nfs-provisioner/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Set up a NFS Server on a Kubernetes cluster
+
+To create a NFS provisioner on your Kubernetes cluster, run the following command
+
+```bash
+kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/nfs-driver-smb/master/examples/kubernetes/nfs-provisioner/nfs-server.yaml
```
Comment (msau42) [inline examples/kubernetes/nfs-provisioner/README.md:6]: This is pointing to smb repo. Should it point to this repo?

File: examples/kubernetes/nfs-provisioner/nfs-server-lb.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                - name: nfs-server
+                  image: gcr.io/kubernetes-e2e-test-images/volume/nfs:1.0
+                  ports:
+                    - containerPort: 2049
+                  volumeMounts:
+                    - mountPath: /nfsshare
+                      name: data-volume
+            volumes:
+                name: data-volume
+                hostPath: /nfsshare-volume
```
Comment (msau42) [inline examples/kubernetes/nfs-provisioner/nfs-server-lb.yaml:41]: This means the nfs server is sticky to that Node right? If the pod gets rescheduled to another node, all the data is lost.

We should probably make it clear of this limitation and say this is for development purposes only.

File: examples/kubernetes/nfs-provisioner/README.md (hunk 3 above)
Comment (mayankshah1607) [reply examples/kubernetes/nfs-provisioner/README.md:6]: Oops, that's a typo

File: examples/kubernetes/nfs-provisioner/README.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Set up a NFS Server on a Kubernetes cluster
+
+> Note: This example is for development perspective only. Because the NFS server is sticky to the node it is scheduled on, data shall be lost if the pod is rescheduled on another node.
+
```
Comment (mayankshah1607) [inline examples/kubernetes/nfs-provisioner/README.md:4]: Thanks @msau42 , I have made the edit here mentioning that the example is for development only

File: examples/kubernetes/nfs-provisioner/nfs-server-lb.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+metadata:
+  name: nfs-server
+  labels:
+    app: nfs-server
+spec:
+  type: LoadBalancer
+  selector:
+    app: nfs-server
+  ports:
+    - port: 2049
```
Comment (andyzhangx) [inline examples/kubernetes/nfs-provisioner/nfs-server-lb.yaml:13]: nit: is there any special meaning of port 2049?

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
/release-note-none

**Pull request merged**